### ISSUE TITLE
TL;DR: Synthesising the Product Engineer Manifesto into 3 simple and accessible questions!

### ISSUE THREAD

**Issue by jpinho** [anttiviljami/product-engineer-manifesto#4]
# Synthesising the Product Engineer Manifesto
_A bi-product of a great sauna session. We did it again 🧖‍♂️ 🧖‍♂️_

The Product Engineer Manifesto, outlines a set of questions that an Engineer should make in order to guarantee he is on the right track to deliver a great and impactful outcome, in exchange for his time and work. The questions listed on the Manifesto, assist the thinking process of the engineer on different angles such as "Understanding/Framing", Craft, Growth and Product Vision. But while these questions are certainly valuable, this list can be quite long. So as an attempt to make this list a bit shorter and "accessible", I've come up with three simple questions: 

### 1. Who is/are the customer/s I am going to deliver this feature to?

- This is important to understand given, the more clients we reach through our investment in the form of work, the more our delivered value will be multiplied, thus increasing our impact.
- We’re in the Software Business and just like in Sports, efficiency is key, we want to do the least amount of effort to achieve the best outcome possible.

- **#be_customer_obsessed #start_with_the_customer**

### 2. What pain (or problem) does it solve? 

- **#know_the_problem #understand_the_pain**

### 3 What is the value and impact of the feature we are delivering?

This question should be asked, considering all aspects: the customer, the business and the engineer track record.

- The higher the value the more likely the customers are to become delighted with our product. It’s great for retention, thus reducing churn rates,  and the closer we get to the fan factor, in which our customers act as our own individual promoters.
- The greater the value the more we can leverage on this feature to grow our MRR, and strengthen our sales speech.
- And lastly, the Engineering Track Record counts too, if as an engineer you’re pitched a feature which has high changes to fail, it super-seeds your skills, or is meaningless on your Track Record, you absolutely must, understand whether it will be wise to commit with delivering such a feature. _You don’t fail, if you don’t start (but you will eventually have to start or your Track Record will suffer), and you also don’t fail until you give up, but you need to make sure you get what it takes to endure, until it’s done._
- **#know_the_impact #measure_the_value**

Aside from these three questions, there is one more which is common sense, but I would like to share it anyways:

**4. How much time is this feature worth investing?**

- Like in everything, time is up the essence. Over a short period of time you want to make sure you deliver the most impactful work you can, that delivers the most value possible to the aforementioned triangle: customer, business, engineer track record.
- Having a timeline, under which you aim to deliver, not only gives you a sense of urgency and commitment, but it also keeps people engaged and focused on your outcomes. E.g: if you say you are going to deliver some feature over the course of 2 years, everyone will disengage and lose interest. Why? It takes too long, you need shorter, meaningful iterations, that continuously ingest customer feedback to keep everyone hooked on the Value-Delivery Hype Train.
- I often see engineers getting lost, into tinkering around, starting to optimise meaningless things, and falling into the pitfall of looking busy as hell, doing the most irrelevant possible work from the Customer & Business point of view. But if it were their company and their money invested, they wouldn’t think twice into taking more moderate and pragmatic decisions. Fair, it may not be your money, but don’t get fooled by that, you have something way more important than money invested. **Your time!** You’ve a very short and limited amount of time to build your Track Record. Which is ultimately your business card to any company, that you are a value-delivery focused professional worth investing on.

## And you know what is the 🍒 (cherry) on top? 

The best of these three questions is that, you don't have to answer them yourself. That would not be wise. But it is unequivocally your responsibility to ask them to the right people.

**1. Who is/are the customer/s I am going to deliver this feature to?** 

- Ask your Product Manager, he will give your the list of customers and the corresponding lead person for contact.

**2. What pain (or problem) does it solve and how can our solution assist on solving that pain?**

- Ask your Product Manager and double check their truth with your Customer Support buddies, Customer Support often has intrinsic knowledge and inside information of the origin of such pains, that can help you better understand the root cause of the pain, enabling you to deliver a more efficient solution. Your Product Manager is a human, and can make mistakes, do your homework and make sure you did your best to ensure you are both right. In the end what counts is that you deliver the right thing, while investing the right amount of time to maximise the value for: the customer, the business and your track record.

**3 What is the value and impact of the feature we are delivering, considering all aspects: the customer, the business and the engineer track record?**

- Ask your Product Manager and double check with Sales if you are not 100% convinced. Asking doesn't hurt, and often sensing how the value of a feature is perceived from the Sales sides can give you a unique perspective over the goal you are about to set yourself to achieve. 

## But why? Why ask these questions?

It is very very simple. These questions represent a **Protocol** that helps you get the information you need in order to build a Story. **Storytelling** is perhaps one of the best skills a Product Engineer can develop. To be able to clearly articulate and communicate who is he shipping a feature to, what pain does it solve, and what's the value and impact of such effort.

👆 If by any means, during the process of telling the story it does not feel right! Then guess what, most likely you should not pursuit it, or you should approach it with a bit more skepticism.

## In a Nutshell 🌰

- Start with the customer(s)
- Identify the pain
- Assess the value & impact
- Gather knowledge from your peers
- Build a story that sounds right or drop it
- Commit & Execute on a reasonable timeframe
- Collect feedback

Regardless of how much you think what you are delivering is important, remember that if you can't fit that into the framework above in a way that makes sense; you are doing it wrong, and it's your fault. Take action, and control the narrative. 

_**"Do it right, Do it light. Do it wrong, Do it long."**;_ this expression emphasises the importance of doing things correctly the first time to avoid having to redo or overcomplicate tasks.

**Comment by jpinho** [comment]
Refinement:  https://www.linkedin.com/pulse/sauna-epiphany-how-i-got-product-engineering-wrong-viljami-kuosmanen-je8be

**Comment by anttiviljami** [comment]
This is now included in the PDF ✅
